### PR TITLE
feat: arrow key navigation for gallery

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/jira/software/c/projects/SFUI2/boards/69?modal=detail&selectedIssue=SFUI2-1127&assignee=5fcf86fca1d3f100684db8a5

# Scope of work

Arrow key navigation for gallery

# Screenshots of visual changes

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/Gallery/GalleryHorizontalArrowKeyNavigation.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  };
+
+  const activeArrowNavigation = (event: React.KeyboardEvent, index: number) => {
+    event.preventDefault();
+    const currentElement = event?.target as HTMLElement;
+    const nextElement = currentElement.nextElementSibling as HTMLElement;
+    const previousElement = currentElement.previousElementSibling as HTMLElement;
+    if ((event.code === 'ArrowRight' || event.code === 'ArrowUp') && index < thumbImages.length - 1) {
+      setActiveIndex(index + 1);
+      nextElement.focus();
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/Gallery/GalleryHorizontalArrowKeyNavigation.tsx:63]: Out of curiosity I checked what would be faster, get all items and target by index or your way, logically I would say your way, but how JS is read by engine might be misleading

And so we are reading from index is faster, when we have read all items, but that would be only accesible if ref would be used on element and then `ref.children` 
image


So at the end your way is simpler and close enough fast to lab testing, however we dont need to use `useRef` which in total with reading item by index, probably would be slower

**Review by Szymon-dziewonski — APPROVED** [review]
good job!

**Review by Szymon-dziewonski — APPROVED** [review]
good job!

File: apps/preview/next/pages/showcases/Gallery/GalleryHorizontalArrowKeyNavigation.tsx (hunk 1 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/Gallery/GalleryHorizontalArrowKeyNavigation.tsx:63]: Nice that you also check such cases :)

**Pull request merged**